src/main/java/net/dv8tion/jda/api/MessageBuilder.java

```diff
@@ -456,6 +461,251 @@ public MessageBuilder replaceLast(@Nonnull String target, @Nonnull String replac
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     */
+    @Nonnull
+    public MessageBuilder clearMentions()
+    {
+        return clearMentionedUsers().clearMentionedRoles();
+    }
+
+    /**
+     * Sets which {@link net.dv8tion.jda.api.entities.Message.MentionType MentionTypes} should be parsed from
+     * the input. This will use {@link MessageAction#getDefaultMentions()} by default, or if {@code null} is provided.
+     *
+     * @param  mentionTypes
+     *         Collection of allowed mention types, or null to use {@link MessageAction#getDefaultMentions()}.
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     */
+    @Nonnull
+    public MessageBuilder setAllowedMentions(@Nullable Collection<Message.MentionType> mentionTypes)
+    {
+        this.allowedMentions = mentionTypes == null
+                ? MessageAction.getDefaultMentions()
+                : Helpers.copyEnumSet(Message.MentionType.class, mentionTypes);
+        return this;
+    }
+
+    /**
+     * Adds the provided {@link net.dv8tion.jda.api.entities.Message.MentionType MentionTypes} to the whitelist.
+     *
+     * @param  types
+     *         The mention types to allow
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     */
+    @Nonnull
+    public MessageBuilder allowMentions(@Nonnull Message.MentionType... types)
+    {
+        Checks.noneNull(types, "MentionTypes");
+        if (types.length > 0)
+        {
+            if (allowedMentions == null)
+                allowedMentions = MessageAction.getDefaultMentions();
+            Collections.addAll(allowedMentions, types);
+        }
+        return this;
+    }
+
+    /**
+     * Removes the provided {@link net.dv8tion.jda.api.entities.Message.MentionType MentionTypes} to the whitelist.
+     *
+     * @param  types
+     *         The mention types to allow
+     *
+     * @throws IllegalArgumentException
+     *         If null is provided
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     */
+    @Nonnull
+    public MessageBuilder denyMentions(@Nonnull Message.MentionType... types)
+    {
+        Checks.noneNull(types, "MentionTypes");
+        if (types.length > 0)
+        {
+            if (allowedMentions == null)
+                allowedMentions = MessageAction.getDefaultMentions();
+            for (Message.MentionType type : types)
+                allowedMentions.remove(type);
+        }
+        return this;
+    }
+
+    /**
+     * Adds the provided {@link IMentionable IMentionable} instance to the whitelist of mentions.
+     * <br>This will only affect instances of {@link User}, {@link Member}, and {@link Role}.
+     *
+     * <p>See {@link MessageAction#mention(IMentionable...)} for more details.
+     *
+     * @param  mentions
+     *         Whitelist of mentions to apply
+     *
```

Comment:
Should this not have a `@throws IllegalArgumentException` considering `denyMentions(Message.MentionType...)` does?